### PR TITLE
(MODULES-3994) Fix incorrect condition so services are created

### PR DESCRIPTION
**Pull request by seanhoughton**
This fixes a typo that prevents the service resources from being created

### CODE REVIEW

**Comment by seanhoughton** [comment]
Hello, I'm reasonably sure that the fix is valid - but I've exhausted my time budget for working out the unit tests today. If I can get some time next week I'll give it another try, but this seems like a fairly serious bug because it results in services not getting installed so maybe someone else can have a look.

**Comment by MosesMendoza** [comment]
@seanhoughton thank you for filing this! I created a ticket, https://tickets.puppetlabs.com/browse/MODULES-3994, to track resolution of this. Agreed this bug needs fixed

**Comment by seanhoughton** [comment]
I'm not sure why the 4.0.0 tests are failing but I'm suspicious of the version comparisons. All the tests pass for me locally running inside a container...

`docker  run --rm -ti -v .:/agent ruby:2.1.10 bash`

File: spec/classes/puppet_agent_spec.rb [hunk 1]
```diff
@@ -109,7 +109,7 @@ def global_facts(facts, os)
           end
         end
 
-        [{}, {:service_names => []}].each do |params|
+        [{}, {}].each do |params|
```
Comment (MikaelSmith) [inline spec/classes/puppet_agent_spec.rb]: Why run this twice now? This was also meant to test with services managed and unmanaged.

**Comment by MikaelSmith** [comment]
Tests are failing. Is that something you're looking into?

**Comment by MikaelSmith** [comment]
Fixed up in https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/209.

**Pull request closed without merging**